### DISCUSSION TITLE
getActivePinia was called with no active Pinia when importing a store outside .vue files

### DISCUSSION THREAD

**Discussion by henrique-arthur** [vuejs/pinia#2465] [Help and Questions] [answered]
The problem is:
I have a .vue file that i'm testing
This .vue imports a .ts file
This .ts file import pinia stores

When i run the tests, occurs this error:
![image](https://github.com/vuejs/pinia/assets/39605748/d39be953-1605-443a-849b-cdd173ae7d2b)

I've solved it by mocking all of the stores i'm importing in .ts file, but i don't want to do that because i need the store in others tests, but if i unmock it the test breaks.

i've tried setActivePinia(createPinia()) and createTestingPinia() but looks like it only works when stores are in SFC files or the store are tested directly on the test file

Can someone help? Thanks for your time :)

**Comment by posva** [comment, marked as answer]
You need to take some measures when using the store outside of components (setup): https://pinia.vuejs.org/core-concepts/outside-component-usage.html#Using-a-store-outside-of-a-component.

The problem you are seeing is that `setActivePinia()` is being called after you import your ts file that runs `useStore()` so it doesn't detect the store.
Don't call `useStore()` directly in a ts file, call it in each component that needs it

**Reply by henrique-arthur** [reply]
Ok, but in this case i need the store in my ts file, there is a way to handle this?

**Reply by posva** [reply]
As the docs point out: don’t call the useStore directly. It can be in a function but it will need the pinia instance as first argument

**Reply by henrique-arthur** [reply]
Just for the record, I found a solution for this, how I need to use stores outside SFC only in one file and my code don't works only on tests, I did this in my .ts file before calling the stores:

```ts
If(process.env.NODE_ENV === 'test'){
  const pinia = createPinia()
  const app = createApp(App)
  app.use(pinia)
}
```
just a turn around to solve the testing problem, until i don't uncouple the code.

@posva thank you for the help 🇧🇷
